assets/pages/new-contributions-landing/components/ContributionForm.jsx

```diff
@@ -150,9 +225,10 @@ function ContributionForm(props: PropTypes) {
         <form
           ref={(el) => {
           if (el) {
-            setupStripe(el, props).then(() => {
-              el.addEventListener('submit', onSubmit);
-            });
+            setupStripe(el, props)
+              .then((stripeHandler) => {
+                el.addEventListener('submit', onSubmit(stripeHandler));
```

Comment:
do we need a `.preventDefault` on this?

Reply:
Yep. I have amended that part. Previously I was using a `ref` which was convenient, but now am using the `onSubmit` property. The side effect is that now the stripe payload is only loaded on form submission, so there is a noticeable delay. I will tackle this in the follow-up, as I need to some housekeeping anyway.